hr_holidays_legal_leave/models/hr_employee.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+# ©  2015 iDT LABS (http://www.@idtlabs.sl)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api, fields
+from openerp.exceptions import Warning as UserError
+
+
+class HrEmployee(models.Model):
+    _inherit = 'hr.employee'
+
+    @api.one
+    def _inverse_remaining_days(self):
+        legal_leave = self.company_id.legal_holidays_status_id
+        if not legal_leave:
+            raise UserError('Legal/annual leave type is not defined for '
+                            'your company.')
+        diff = self.remaining_leaves - legal_leave.get_days(
+            self.id)[legal_leave.id]['remaining_leaves']
+        if diff > 0:
+            leave = self.env['hr.holidays'].create(
+                {
+                    'name': 'Allocation for %s' % self.name,
+                    'employee_id': self.id,
+                    'holiday_status_id': legal_leave.id,
+                    'type': 'add',
+                    'holiday_type': 'employee',
+                    'number_of_days_temp': diff
+                }
+            )
+        elif diff < 0:
+            raise UserError(
+                'You cannot reduce validated allocation requests.')
+
+        for sig in ('confirm', 'validate', 'second_validate'):
+            leave.signal_workflow(sig)
+
+    @api.one
+    def _compute_remaining_days(self):
```

Comment:
From my experience, you end up needing to manually modify the remaining days. You could add this feature idea to the roadmap.